mcf-browser/src/TimelineExplorer.tsx

```diff
@@ -0,0 +1,492 @@
+    * @param {string[]} locations a list of locations to filter by
+    * @return {Grouping} an object mapping from variableMeasured to an array
+    * of Series with that variableMeasured value
+    */
+  getVariableToSeries(locations: string[]): Grouping {
+    const output: Grouping = {};
+    // Filter by location
+    const data = this.getFilteredData(locations);
+
+    for (const series of data) {
+      const varMeasured = series.variableMeasured ?
+        series.variableMeasured :
+        '';
+      if (!output[varMeasured]) {
+        output[varMeasured] = [];
+      }
+      output[varMeasured].push(series);
+    }
+
+    return output;
+  }
+
+
+  /** Returns the JSX to render a group of related series
+    * @param {Series[]} seriesList a list of series objects with
+    *                              the same varMeasured
+    * @param {LocationMapping} locationMapping a mapping from location dcid to
+    * name
+    * @return {JSX.Element} a details element plotting all of the series
+   */
+  renderSeriesGroup(seriesList: Series[], locationMapping: LocationMapping)
+    : JSX.Element | null {
+    if (seriesList.length === 0) {
+      return null;
+    }
+    const varMeasured = seriesList[0].variableMeasured;
+    const groups = this.groupSeriesByLocations(seriesList);
+    const groupNames = Object.keys(groups);
+
+    return (
+      <details className="stat-var-section" key={varMeasured}>
+        <summary>{varMeasured}</summary>
+        {
+          groupNames.map(
+              (groupName) =>
+                this.renderTimeGraph(
+                    groups[groupName],
+                    groupName,
+                    groupNames.length === 1,
+                    locationMapping,
+                ),
+          )
+        }
+      </details>
+    );
+  }
+
+  /** Get all locations using the observationAbout property
+    * @return {SelectOption[]} a list of unique location objects
+    */
+  private async getAllLocations(): Promise<SelectOption[]> {
+    // Get an array of unique locations
+    const seenLocations = new Set<string>();
+    const locations: string[] = [];
+    for (const series of this.props.data) {
+      const location = series.observationAbout;
+      if (!seenLocations.has(location)) {
+        locations.push(location);
+        seenLocations.add(location);
+      }
+    }
+
+    const labelPromises = locations.map((location) => {
+      const label =
+        (location.startsWith('dcid:') ?
+          getName(location.slice(5)) :
+          getName(location)
+        );
+      return label;
+    });
+
+    return Promise.all(labelPromises).then(
+        (labels) => {
+          return locations.map((location, i) => {
+            return {
+              value: location,
+              label: labels[i],
+            };
+          });
+        },
+    );
+  }
+
+
+  /** Groups data with equal values for everything except for their
+      * locations into groups of groupNumber to be plotted together
```

Comment:
i believe this indentation is off. Please check the indentation of function comments in this pr, should be: 

```
/**
 *
 *
 */
```